itests/kit/node_unmanaged.go

```diff
@@ -344,11 +344,121 @@ func (tm *TestUnmanagedMiner) OnboardSectorWithPiecesAndMockProofs(ctx context.C
 
 	respCh := make(chan WindowPostResp, 1)
 
-	go tm.wdPostLoop(ctx, sectorNumber, respCh, true)
+	go tm.wdPostLoop(ctx, sectorNumber, respCh, true, tm.sealedCids[sectorNumber], tm.sealedSectorPaths[sectorNumber], tm.cacheDirPaths[sectorNumber])
 
 	return sectorNumber, respCh
 }
 
+func (tm *TestUnmanagedMiner) mkStagedFileWithPieces(_ context.Context, sectorNumber abi.SectorNumber, pt abi.RegisteredSealProof) ([]abi.PieceInfo, string) {
+	paddedPieceSize := abi.PaddedPieceSize(tm.options.sectorSize)
+	unpaddedPieceSize := paddedPieceSize.Unpadded()
+
+	// Generate random bytes for the piece
+	randomBytes := make([]byte, unpaddedPieceSize)
+	_, err := io.ReadFull(rand.Reader, randomBytes)
+	require.NoError(tm.t, err)
+
+	// Create a temporary file for the first piece
+	pieceFileA := requireTempFile(tm.t, bytes.NewReader(randomBytes), uint64(unpaddedPieceSize))
+
+	// Generate the piece CID from the file
+	pieceCIDA, err := ffi.GeneratePieceCIDFromFile(pt, pieceFileA, unpaddedPieceSize)
+	require.NoError(tm.t, err)
+
+	// Reset file offset to the beginning after CID generation
+	_, err = pieceFileA.Seek(0, io.SeekStart)
+	require.NoError(tm.t, err)
+
+	unsealedSectorFile := requireTempFile(tm.t, bytes.NewReader([]byte{}), 0)
+	defer func() {
+		_ = unsealedSectorFile.Close()
+	}()
+
+	// Write the piece to the staged sector file without alignment
+	writtenBytes, pieceCID, err := ffi.WriteWithoutAlignment(pt, pieceFileA, unpaddedPieceSize, unsealedSectorFile)
+	require.NoError(tm.t, err)
+	require.EqualValues(tm.t, unpaddedPieceSize, writtenBytes)
+	require.True(tm.t, pieceCID.Equals(pieceCIDA))
+
+	// Create a struct for the piece info
+	publicPieces := []abi.PieceInfo{{
+		Size:     paddedPieceSize,
+		PieceCID: pieceCIDA,
+	}}
+
+	return publicPieces, unsealedSectorFile.Name()
+}
+
+func (tm *TestUnmanagedMiner) SnapDealWithRealProofs(ctx context.Context, proofType abi.RegisteredSealProof, sectorNumber abi.SectorNumber) chan WindowPostResp {
+	// generate sector key
+	pieces, unsealedPath := tm.mkStagedFileWithPieces(ctx, sectorNumber, proofType)
+	updateProofType := abi.SealProofInfos[proofType].UpdateProof
+
+	s, err := os.Stat(tm.sealedSectorPaths[sectorNumber])
+	require.NoError(tm.t, err)
+
+	randomBytes := make([]byte, s.Size())
+	_, err = io.ReadFull(rand.Reader, randomBytes)
+	require.NoError(tm.t, err)
+
+	updatePath := requireTempFile(tm.t, bytes.NewReader(randomBytes), uint64(s.Size()))
+	_ = updatePath.Close()
+	updateDir := filepath.Join(tm.t.TempDir(), fmt.Sprintf("update-%d", sectorNumber))
+	_ = os.MkdirAll(updateDir, 0700)
```

Comment:
can we check these two errors?
```suggestion
	require.NoError(tm.t, updatePath.Close())
	updateDir := filepath.Join(tm.t.TempDir(), fmt.Sprintf("update-%d", sectorNumber))
	require.NoError(tm.t, os.MkdirAll(updateDir, 0700))
```